eternalcore-core/src/main/java/com/eternalcode/core/feature/jail/JailServiceImpl.java

```diff
@@ -0,0 +1,331 @@
+            return;
+        }
+
+        Prisoner prisoner = new Prisoner(player.getUniqueId(), Instant.now(), time, detainedBy.getUniqueId());
+
+        if (isPlayerJailed) {
+            this.prisonersRepository.editPrisoner(prisoner);
+        }
+        else {
+            this.prisonersRepository.savePrisoner(prisoner);
+        }
+
+        this.jailedPlayers.put(player.getUniqueId(), prisoner);
+
+        Location jailLocation = PositionAdapter.convert(this.jailPosition);
+
+        this.teleportService.teleport(player, jailLocation);
+
+        this.noticeService.create()
+            .notice(translation -> translation.jailSection().jailDetainPublic())
+            .placeholder("{PLAYER}", player.getName())
+            .all()
+            .send();
+
+        this.noticeService.create()
+            .notice(translation -> translation.jailSection().jailDetainPrivate())
+            .player(player.getUniqueId())
+            .send();
+
+    }
+
+
+    @Override
+    public void releasePlayer(Player player, @Nullable Player releasedBy) {
+
+        if (releasedBy != null) {
+            if (!this.jailedPlayers.containsKey(player.getUniqueId())) {
+                this.noticeService.create()
+                    .notice(translation -> translation.jailSection().jailReleaseNoPlayer())
+                    .player(releasedBy.getUniqueId())
+                    .send();
+                return;
+            }
+
+            this.noticeService.create()
+                .notice(translation -> translation.jailSection().jailReleaseSender())
+                .placeholder("{PLAYER}", player.getName())
+                .player(releasedBy.getUniqueId())
+                .send();
+        }
+
+        JailReleaseEvent jailReleaseEvent = new JailReleaseEvent(player.getUniqueId());
+
+        if (jailReleaseEvent.isCancelled()) {
+            return;
+        }
+
+        this.prisonersRepository.deletePrisoner(player.getUniqueId());
+        this.jailedPlayers.remove(player.getUniqueId());
+
+        this.spawnService.teleportToSpawn(player);
+
+        this.noticeService.create()
+            .notice(translation -> translation.jailSection().jailReleasePrivate())
+            .player(player.getUniqueId())
+            .send();
+
+        this.noticeService.create()
+            .notice(translation -> translation.jailSection().jailReleasePublic())
+            .placeholder("{PLAYER}", player.getName())
+            .all()
+            .send();
+
+    }
+
+    @Override
+    public void releaseAllPlayers(Player player) {
+
+        if (this.jailedPlayers.isEmpty()) {
+            this.noticeService.create()
+                .notice(translation -> translation.jailSection().jailReleaseNoPlayers())
+                .player(player.getUniqueId())
+                .send();
+            return;
+        }
+
+        this.jailedPlayers.forEach((uuid, prisoner) -> {
+                Player jailedPlayer = Bukkit.getPlayer(uuid);
```

Comment:
Store server variable, and use it to get the player
```suggestion
                Player jailedPlayer = this.server.getPlayer(uuid);
```